src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -17,33 +17,43 @@
     private ShoppingCartDao shoppingCartDao;
     @Inject
     private TicketDao ticketDao;
+    @Inject
+    private UserDao userDao;
 
     @Override
     public void addSession(MovieSession movieSession, User user) {
-        Ticket newTicket = new Ticket();
-        newTicket.setUser(user);
-        newTicket.setMovieSession(movieSession);
-
-        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
-        shoppingCart.getTickets().add(ticketDao.add(newTicket));
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCart = getByUser(user);
+        shoppingCart.getTickets().add(ticket);
         shoppingCartDao.update(shoppingCart);
     }
 
     @Override
     public ShoppingCart getByUser(User user) {
-        return shoppingCartDao.getByUser(user);
+        return shoppingCartDao.getByUser(user).orElseThrow(
+                () -> new RuntimeException("ShoppingCart not found")
+        );
     }
 
     @Override
     public void registerNewShoppingCart(User user) {
+        if (userDao.get(user.getEmail()).isEmpty()) {
+            throw new RuntimeException("There is no user: " + user);
+        }
+        if (shoppingCartDao.getByUser(user).isPresent()) {
+            throw new RuntimeException("ShoppingCart already exists: " + user);
```

Comment:
Consider using a more specific exception type instead of `RuntimeException` for better error handling and clarity.